### PR TITLE
Codegen: Move generated data into workspace.c

### PR DESCRIPTION
**Pull request by imciner2**
Modify the code generation routines to place the data into a workspace.c file and create prototypes for the variables (using extern) in workspace.h.

This allows for workspace.h to be included in multiple object files that are then linked together. Previously, the linker would error due to multiple definitions of the data variables (one in each object).

### CODE REVIEW

File: codegen/render_workspace.m [hunk 1]
```diff
@@ -1,51 +1,64 @@
...
-fprintf(f, '#define %s\n\n', incGuard);
+fprintf(incFile, '#ifndef %s\n', incGuard);
+fprintf(incFile, '#define %s\n\n', incGuard);
 
 % Include types, constants and private header
-fprintf(f, '#include \"types.h\"\n');
-fprintf(f, '#include \"qdldl.h\"\n\n');
-fprintf(f, '#include \"qdldl_interface.h\"\n');
+fprintf(incFile, '#include \"types.h\"\n');
+fprintf(incFile, '#include \"qdldl.h\"\n');
```
Comment (gbanjac) [inline codegen/render_workspace.m]: I think we don't need to include `qdldl.h`. I know this was here earlier, but maybe you can just remove it.

File: codegen/render_workspace.m [hunk 2]
```diff
@@ -1,51 +1,64 @@
...
+fprintf(incFile, '#include \"qdldl_interface.h\"\n\n');
+
+fprintf(srcFile, '#include \"types.h\"\n');
+fprintf(srcFile, '#include \"qdldl.h\"\n');
```
Comment (gbanjac) [inline codegen/render_workspace.m]: same here

**Comment by gbanjac** [comment]
It looks very good! Thanks a lot for doing this. I have just minor comments.

**Comment by imciner2** [comment]
Yep, `qldl.h` is not needed in the generated source files. It has been removed from the generator now.

File: codegen/render_workspace.m [hunk 3]
```diff
@@ -261,3 +404,14 @@ function write_mat(f, mat, name)
 fprintf(f, '%d};\n', mat.nz);
 
 end
+
+function write_mat_extern(f, mat, name)
+%WRITE_MAT_EXTERN Write matrix the prototype for the matrix.
+
+write_vec_extern(f, mat.i, [name, '_i'], 'c_int');
+write_vec_extern(f, mat.p, [name, '_p'], 'c_int');
+write_vec_extern(f, mat.x, [name, '_x'], 'c_float');
```
Comment (gbanjac) [inline codegen/render_workspace.m]: I removed the three lines above in https://github.com/oxfordcontrol/osqp-matlab/commit/cee47d25218695e1c0b5390f7c7c8330d387f1b9

**Pull request merged**